app/room.py

```diff
@@ -0,0 +1,129 @@
+from abc import abstractmethod
+from person import Fellow, Staff
+
+
+class Room(object):
+    """docstring for Room."""
+
+    def __init__(self, name):
+        if not name:
+            raise TypeError('name is required')
+        self.name = name
+
+    # to be implemented by all sub classes
+    @abstractmethod
+    def is_vacant(self):
+        pass
+
+
+class Office(Room):
+    """docstring for Office."""
```

Comment:
This is not how you write a function doc string. Please put a better description than what is there.